lib/ansible/modules/monitoring/grafana_dashboard.py

```diff
@@ -0,0 +1,317 @@
+version_added: "2.5"
+short_description: Manage grafana dashboards
+description:
+  - Create/update/delete/export grafana dashboards via API
+options:
+  grafana_url:
+    required: true
+    description:
+      - Grafana url
+  grafana_user:
+    required: false
+    default: admin
+    description:
+      - Grafana API user
+  grafana_password:
+    required: false
+    default: admin
+    description:
+      - Grafana API password
+  grafana_api_key:
+    required: false
+    description:
+      - Grafana API key
+      - If set, grafana_user and grafana_password will be ignored
+  org_id:
+    required: false
+    description:
+      - Grafana Organisation ID where the dashboard will be imported / exported
+      - Not used when grafana_api_key is set, because the grafana_api_key only belong to one organisation.
+    default: 1
+  state:
+    required: true
+    default: present
+    description:
+      - State of the dasboard.
+      - can be 'present', 'absent', 'export'
+  slug:
+    description:
+      - slug of the dashboard. It's the friendly url name of the dashboard.
+      - When state is present, this parameter can override the slug in the meta section of the json file.
+      - If you want to import a json dashboard exported directly from the interface (not from the api),
+      - you have to specify the slug parameter because there is no meta section in the exported json.
+  path:
+    description:
+      - path to the json file contaning the grafana dashboard to import or export.
+  overwrite:
+    default: false
+    description:
+      - override existing dashboard when state is present.
+  message:
+    description:
+      - Set a commit message for the version history.
+      - Only used when state is present
+  validate_cert:
+    default: false
+    description:
+      - trust or not the grafana api tls certificate.
+'''
+
+EXAMPLES = '''
+---
+- name: import grafana dashboard foo
+  grafana_dashboard:
+    grafana_url: http://grafana.company.com
+    grafana_api_key: XXXXXXXXXXXX
+    state: present
+    message: "updated by ansible"
+    overwrite: true
+    path: /path/to/dashboards/foo.json
+
+- name: export dashboard
+  grafana_dashboard:
+    grafana_url: http://grafana.company.com
+    grafana_api_key: XXXXXXXXXXXX
+    state: export
+    slug: foo
+    path: /path/to/dashboards/foo.json
+'''
+
+RETURN = '''
+---
+slug:
+  description: slug of the created / deleted / exported dashboard.
+  returned: success
+  type: string
+  sample: foo
+'''
+
+import base64
+import json
+import os
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url
+
+__metaclass__ = type
+
+
+class GrafanaAPIException(Exception):
+    pass
+
+
+class GrafanaMalformedJson(Exception):
+    pass
+
+
+def grafana_switch_organisation(module, grafana_url, org_id, headers):
+    r, info = fetch_url(module, '{}/api/user/using/{}'.format(grafana_url, org_id), headers=headers, method='POST')
```

Comment:
Uses of `{}` should be updated with positions as this will be incompatible with python 2.6.  This applies through the code base.

e.g. `'{0}'.format('foo')`